pkg/compute/executor.go

```diff
@@ -302,8 +302,10 @@ func (e *BaseExecutor) Run(ctx context.Context, state store.LocalExecutionState)
 	stopwatch := telemetry.Timer(ctx, jobDurationMilliseconds, state.Execution.Job.MetricAttributes()...)
 	topic := EventTopicExecutionRunning
 	defer func() {
-		if err != nil && err.Error() != executor.ErrAlreadyCancelled.Error() {
-			e.handleFailure(ctx, state, err, topic)
+		if err != nil {
+			if baseErr, ok := err.(*models.BaseError); ok && baseErr.Code() == executor.ExecutionAlreadyCancelled {
+				e.handleFailure(ctx, state, baseErr, topic)
```

Comment:
should be `!=`. Some helper methods in `models` might simplify the error checks, such as 
```
// IsErrorWithCode checks if the given error is a BaseError (or its pointer)
// and if its code matches any of the provided expected codes.
func IsErrorWithCode(err error, expectedCodes ...ErrorCode) bool {
	var baseErr *BaseError
	if errors.As(err, &baseErr) {
		errCode := baseErr.Code()
		for _, code := range expectedCodes {
			if errCode == code {
				return true
			}
		}
	}
	return false
}
```

and then you can say:
```
	if !IsErrorWithCode(err, executor.ExecutionAlreadyCancelled) {
		e.handleFailure(ctx, state, err, topic)
	}
```

Reply:
Done